src/common/core-functions.k

```diff
@@ -109,14 +139,35 @@ syntax KItem ::=  mapDiff(
                 K   //setWrap(Set)
               )
               [strict]
-
-rule mapDiff( mapWrap( (Key:K |-> _ => .Map) _), setWrap( (SetItem(Key) => .Set) _) )
-
-rule mapDiff( mapWrap( MyMap:Map ), setWrap( (SetItem(Key:K) => .Set) _) )
-when
-    notBool Key in keys(MyMap)
-
-rule mapDiff(mapWrap( MyMap:Map ), setWrap(.Set)) => mapWrap(MyMap)
+//todo
+//rule mapDiff( mapWrap( (Key:K |-> _ => .Map) _:Map), setWrap( (SetItem(Key) => .Set) _:Set) )
+//rule mapDiff( mapWrap( MyMap:Map ), setWrap( (SetItem(Key:K) => .Set) _:Set) )
+//when    notBool Key in keys(MyMap)
+//rule mapDiff(mapWrap( MyMap:Map ), setWrap(.Set)) => mapWrap(MyMap)
+
+//@ Substracts from the given map the value given (match is on the value)
+syntax KItem ::= removeMapItemsByValue (Map,K)             [function]
```

Comment:
If not possible to get rid of the K-related sorts from the definition due to K limitations, then put a note about that somewhere, so we know to come back to this issue later.  But in the long run we'd like our users to not even be aware of sorts like K, KItem, KList, etc., unless they define meta-level things.